zcash_local_net/src/validator.rs

```diff
@@ -1,151 +1,35 @@
-///
-/// If `chain_cache` path is `None`, a new chain is launched.
-pub struct ZcashdConfig {
-    /// Zcashd RPC listen port
-    pub rpc_listen_port: Option<Port>,
-    /// Local network upgrade activation heights
-    pub configured_activation_heights: testnet::ConfiguredActivationHeights,
-    /// Miner address
-    pub miner_address: Option<&'static str>,
-    /// Chain cache path
-    pub chain_cache: Option<PathBuf>,
-}
-
-impl ZcashdConfig {
-    /// Regtest-friendly defaults for testing.
-    pub fn default_test() -> Self {
-        Self {
-            rpc_listen_port: None,
-            configured_activation_heights: for_test::all_height_one_nus(),
-            miner_address: Some(REG_O_ADDR_FROM_ABANDONART),
-            chain_cache: None,
-        }
-    }
-}
-
-/// Zebrad configuration
-///
-/// Use `zebrad_bin` to specify the binary location.
-/// If the binary is in $PATH, `None` can be specified to run "zebrad".
-///
-/// If `rpc_listen_port` is `None`, a port is picked at random between 15000-25000.
-///
-/// Use `activation_heights` to specify custom network upgrade activation heights.
-///
-/// Use `miner_address` to specify the target address for the block rewards when blocks are generated.
-///
-/// If `chain_cache` path is `None`, a new chain is launched.
-///
-/// `network` can be used for testing against cached testnet / mainnet chains where large chains are needed.
-/// `activation_heights` and `miner_address` will be ignored while not using regtest network.
-#[derive(Clone)]
-pub struct ZebradConfig {
-    /// Zebrad network listen port
-    pub network_listen_port: Option<Port>,
-    /// Zebrad JSON-RPC listen port
-    pub rpc_listen_port: Option<Port>,
-    /// Zebrad gRPC listen port
-    pub indexer_listen_port: Option<Port>,
-    /// Local network upgrade activation heights
-    pub configured_activation_heights: testnet::ConfiguredActivationHeights,
-    /// Miner address
-    pub miner_address: &'static str,
-    /// Chain cache path
-    pub chain_cache: Option<PathBuf>,
-    /// Network type
-    pub network: NetworkKind,
-}
-
-impl ZebradConfig {
-    /// Zebrad defaults for testing
-    pub fn default_test() -> Self {
-        Self {
-            network_listen_port: None,
-            rpc_listen_port: None,
-            indexer_listen_port: None,
-            configured_activation_heights: for_test::all_height_one_nus(),
-            miner_address: ZEBRAD_DEFAULT_MINER,
-            chain_cache: None,
-            network: NetworkKind::Regtest,
-        }
-    }
+use zcash_protocol::PoolType;
+use zebra_chain::parameters::testnet;
+use zebra_chain::parameters::testnet::ConfiguredActivationHeights;
+use zebra_chain::parameters::NetworkKind;
+
+use crate::process::IsAProcess;
+
+pub mod zcashd;
+pub mod zebrad;
+
+/// Can offer specific functionality shared across configuration for all validators.
+pub trait ValidatorConfig: Default {
+    /// To set up some stuff for Regtest.
```

Comment:
please can we maintain a professional standard for comments i.e.
```suggestion
    /// Sets the config for common Regtest parameters.
```

Also the issue here is that zcash_local_net should not enforce which seed the faucet is associated with so i'm not sure how `mine_to_pool` is going to work. I will read on, but i assume we now have abandon art addresses in zcash_local_net where it was designed to be very generic and flexible for external consumers also. The mining address is set in zingolib and zingolib dictates that abandon art is the faucet

Reply:
on reflection we can just document clearly that abandon art is the faucet for all consumers